bin/gaffer

```diff
@@ -268,6 +268,30 @@ if [[ -n $DELIGHT ]] ; then
 
 fi
 
+# Set up 3rd Party extensions
+##########################################################################
+
+for extension in `echo $GAFFER_EXTENSION_PATHS | sed 's/:/\n/g'`; do
+
+	echo Loading extension: $extension
+
+	if [[ `uname` = "Linux" ]] ; then
+		appendToPath "$extension/lib" LD_LIBRARY_PATH
+	else
+		appendToPath "$extension/lib" DYLD_LIBRARY_PATH
+	fi
+
+	appendToPath "$extension/bin" PATH
+	appendToPath "$extension/python" PYTHONPATH
+	appendToPath "$extension/startup" GAFFER_STARTUP_PATHS
```

Comment:
I don't think were getting the order we'd like here :

```
> GAFFER_EXTENSION_PATHS="/test" gaffer env | grep GAFFER_STARTUP_PATHS
GAFFER_STARTUP_PATHS=/home/john/gaffer/startup:/disk1/john/dev/build/gaffer/startup:/test/startup
# (/disk1/john/dev/build/gaffer is my gaffer installation)
```

We'd want the extension path to come before the install path, so it can override things done by the standard install. I suspect similar things might apply in the other cases too.

Reply:
You’re right, startup should prepend instead.

Which others do you think should prepend as well? I wouldn’t want an extension to prepend libs (in case one inadvertently supplied a dependency lib in the same folder).

Reply:
That's a tough question, and I think it boils down to how much we want to trust extension authors. How about we start conservatively and just do GAFFER_STARTUP_PATHS?